### ISSUE TITLE
Descheduling plugin examples with guides

### ISSUE THREAD

**Issue by ingvagabund** [kubernetes-sigs/descheduler#923]
As part of implementing new descheduling plugins it is important to provide examples with basic building blocks for various use cases. The goal of the issue is to provide guidance on how to build a plugin and how to integrate it with the descheduling framework.

Examples of descheduling plugins can be already provided since all the existing strategies have been migrated to plugins: https://github.com/kubernetes-sigs/descheduler/tree/master/pkg/framework/plugins. The plugin arguments will get moved to their corresponding plugins so the guide will need to be updated if the guide is provided before hand.

Once the plugin registry is implemented as well the guides can be extended with instructions on how to integrate a plugin into the framework.

**Comment by pravarag** [comment]
I would like to take this one and finish it at the earliest :)

/assign

**Comment by damemi** [comment]
What will be great about this is, once it's done, this whole repo will itself be an implementation of the framework (ie, the default Descheduler). So we should use and reference that in our guides.

Still, smaller examples will also help to show abstract uses of the individual concepts in the framework. I think a balance of both will be most useful.

**Comment by pravarag** [comment]
> What will be great about this is, once it's done, this whole repo will itself be an implementation of the framework (ie, the default Descheduler). So we should use and reference that in our guides.
> 
> Still, smaller examples will also help to show abstract uses of the individual concepts in the framework. I think a balance of both will be most useful.

Agreed. I was thinking about starting small, by keeping a single document under [docs](https://github.com/kubernetes-sigs/descheduler/tree/master/docs) for now and add examples in it. Since all the plugins are available as shared by @ingvagabund above, @damemi is there any preference for cluster while capturing the examples? I use Kind locally (since it supports multinode clusters as well).

**Comment by ingvagabund** [comment]
Hi @pravarag. Do you still plan on taking this item?

**Comment by ricardomaraschini** [comment]
I have written a very simple plugin [here|https://github.com/kubernetes-sigs/descheduler/issues/923], can you please take a look ?

**Comment by ingvagabund** [comment]
Resolved through https://github.com/kubernetes-sigs/descheduler/pull/1636.
